IPython/core/compilerop.py

```diff
@@ -92,6 +92,9 @@ def __init__(self):
         # (otherwise we'd lose our tracebacks).
         linecache.checkcache = check_linecache_ipython
 
+        # Caching a dictionary { filename: execution_count } for nicely
+        # rendered tracebacks
+        self.filename_map = {}
```

Comment:
Let's just take steps to ensure this dict won't get used somewhere else. Also, `filename_map` is a bit misleading here -  these aren't filenames after all ;).

```suggestion
        self._co_filename_map = {}
```

Reply:
Filename here refers to the `filename` argument passed to the [compile function](https://docs.python.org/3/library/functions.html#compile) used for AST compilation.

I agree, let's make it more private.

Reply:
If I saw this, it would be easier to connect the dots if it told me that it's related to `co_filename`, but it's probably just me having done one too many frame hacks in python.

Up to you :).